src/esp/metadata/attributes/LightLayoutAttributes.h

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+
+#ifndef ESP_METADATA_ATTRIBUTES_LIGHTLAYOUTATTRIBUTES_H_
+#define ESP_METADATA_ATTRIBUTES_LIGHTLAYOUTATTRIBUTES_H_
+
+#include "AttributesBase.h"
+
+namespace esp {
+namespace metadata {
+namespace attributes {
+
+/**
+ * @brief This class describes an instance of a light -
+ * it's template name, location/direction, color, intensity, type and other
+ * parameters if appropriate.
+ */
+class LightInstanceAttributes : public AbstractAttributes {
+ public:
+  LightInstanceAttributes(const std::string& handle = "");
+
+  /**
+   * @brief Get/Set the position of the light.
+   */
+  void setPosition(const Magnum::Vector3& position) {
+    setVec3("position", position);
+  }
+  Magnum::Vector3 getPosition() const { return getVec3("position"); }
+
+  /**
+   * @brief Get/Set the direction of the light.
+   */
+  void setDirection(const Magnum::Vector3& direction) {
+    setVec3("direction", direction);
+  }
+  Magnum::Vector3 getDirection() const { return getVec3("direction"); }
+
+  /**
+   * @brief Get/Set the color of the light.
+   */
+  void setColor(const Magnum::Vector3& color) { setVec3("color", color); }
+  Magnum::Vector3 getColor() const { return getVec3("color"); }
+
+  /**
+   * @brief Get/Set the color scale of the light.
+   */
+  void setIntensity(double intensity) { setDouble("intensity", intensity); }
+  double getIntensity() const { return getDouble("intensity"); }
+
+  /**
+   * @brief Get/Set the type of the light
+   */
+  void setType(const std::string& type) { setString("type", type); }
+  std::string getType() const { return getString("type"); }
+
+  /**
+   * @brief Get/Set inner cone angle for spotlights.  Should be ignored for
+   * other lights
+   */
+  void setInnerConeAngle(double innerConeAngle) {
+    setDouble("innerConeAngle", innerConeAngle);
+  }
+  double getInnerConeAngle() const { return getDouble("innerConeAngle"); }
```

Comment:
Is this radians or degrees? :) Why not use Magnum's angle types as I do in [LightData](https://doc.magnum.graphics/magnum/classMagnum_1_1Trade_1_1LightData.html)? These are exposed to Python, have the dedicated `_rad(f)` / `_deg(f)` literals etc. so the usage should be rather painless. It also means that no matter whether the actual JSON has radians or degrees, you can still do stuff like `setInnerConeAngle(15.0_degf)` without having to convert manually.

Reply:
That sounds cool, I'll investigate further. Thanks!